src/odemis/dataio/tiff.py

```diff
@@ -495,7 +495,23 @@ def _convertToOMEMD(images, multiple_files=False, findex=None, fname=None, uuids
             description.text = da0.metadata[model.MD_HW_NOTE]
 
         # light source must come before detector for valid ome-xml
-        if (model.MD_LIGHT_POWER in da0.metadata or
+        # meteor multi-channel lightsource
+        if model.MD_LIGHT_POWER in da0.metadata and (
+            model.MD_IN_WL in da0.metadata or model.MD_OUT_WL in da0.metadata
+        ):
+            # TODO: what is the best way to check if image is meteor?
+            # add a light source for each channel, as they can have different power
+            for gdid, gr in enumerate(g):
+
+                obj = ET.Element("LightEmittingDiode",
+                                    attrib={"ID": "LightSource:%d" % gdid})
```

Comment:
Using just the gdid here is incorrect, as there can be several groups of zstacks. So, something like `did + gdid` would more likely work in every case.